src/ports/referral/component.ts

```diff
@@ -0,0 +1,38 @@
+import { isErrorWithMessage } from '../../logic/error-handling'
+import { AppComponents } from '../../types'
+import { IReferralComponent, ReferralError } from './types'
+
+export async function createReferralComponent({ config, logs }: Pick<AppComponents, 'config' | 'logs'>): Promise<IReferralComponent> {
+  const logger = logs.getLogger('referral-server')
+  const referralServerUrl = await config.requireString('REFERRAL_SERVER_URL')
+  const apiKey = await config.requireString('REFERRAL_SERVER_API_KEY')
```

Comment:
As these are required on component creation, please add them to the definitions in order to ensure a proper service start.